src/daft-core/src/utils/supertype.rs

```diff
@@ -191,7 +191,7 @@ pub fn get_supertype(l: &DataType, r: &DataType) -> Option<DataType> {
             // }
 
             // every known type can be casted to a string except binary
-            (dt, DataType::Utf8) if !matches!(&dt, &DataType::Binary | &DataType::FixedSizeBinary(_)) => Some(DataType::Utf8),
+            (dt, DataType::Utf8) if !matches!(&dt, &DataType::Binary | &DataType::FixedSizeBinary(_) | &DataType::List(_)) => Some(DataType::Utf8),
```

Comment:
this is somewhat unrelated, but I found that list to utf8 casting currently does not work, so we error out when attempting to get the supertype instead of panicking at runtime.